docs/Using-Push-Service.md

```diff
@@ -66,11 +76,32 @@ FirebaseInstanceId.getInstance().instanceId.addOnCompleteListener { task ->
 }
 ```
 
-## Push Service API Reference
+_To be able to successfully process notifications, you need to register the app to receive push notifications in the first place. For more information visit [official documentation](https://firebase.google.com/docs/cloud-messaging/android/client)._
 
-All available methods of the `IPushService` API are:
+## Receiving WMT Push Notifications
 
-- `acceptLanguage` - Language settings, that will be sent along with each request.
-- `register(fcmToken: String, listener: IPushRegisterListener)` - Registers Firebase Cloud Messaging token on the backend
-    - `fcmToken` - Firebase Cloud Messaging token.
-    - `listener` - Called when the request finishes.
\ No newline at end of file
+To process the raw notification obtained from Firebase Cloud Messaging service (FCM), you can use `PushParser` helper class that will parse the notification into a `PushMessage` result.
+
+The `PushMessage` is an abstract class that is implemented by following classes for concrete results
+
+- `PushMessageOperationCreated` - a new operation was triggered with the following properties
+  -  `id` of the operation
+  -  `name` of the operation
+- `PushMessageOperationFinished` - an operation was finished, successfully or non-successfully with following properties
+  -  `id` of the operation
+  -  `name` of the operation
+  -  `result` of the operation (for example that the operation was canceled by the user).
+
+
+_Example push notification processing:_
+
+```kotlin
+// Overriden method of FirebaseMessagingService
+override fun onMessageReceived(remoteMessage: RemoteMessage) {
+    super.onMessageReceived(remoteMessage)
+    val push = PushParser.parseNotification(remoteMessage.data)
+    if (push != null) {
+        // process the notification and react to it in the UI
+    }
```

Comment:
Add info about the fact that this is the place where the rest of the processing should happen, i.e.:

```kotlin
    // process the mtoken notification and react to it in the UI
} else {
    // process all the other notification types using your own logic
}

Reply:
👍